conans/client/generators/text.py

```diff
@@ -65,7 +65,7 @@ def _loads_deps_user_info(text):
 
     @staticmethod
     def _loads_cpp_info(text):
-        pattern = re.compile("^\[([a-zA-Z0-9_:-]+)\]([^\[]+)", re.MULTILINE)
+        pattern = re.compile("^\[([a-zA-Z0-9_:-^.]+)\]([^\[]+)", re.MULTILINE)
```

Comment:
This RE is not good. It will allow "[includedirs_My_Other^Lib]" too. `:-^.` is any character between ^ and . ascii codes.
The right ER could be:

`re.compile(r"^\[([a-zA-Z0-9_:-\\.]+)\]([^\[]+)", re.MULTILINE)`